docs/standard/native-interop/calling-conventions.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Unmanaged calling conventions
+description: Learn how to specify unmanaged calling conventions in .NET
+author: jkotas
+ms.date: 08/19/2023
+---
+
+# Unmanaged calling conventions
+
+[Calling conventions](https://en.wikipedia.org/wiki/Calling_convention) desribe low-level details for how method arguments and return values
+are passed between the caller and the called method.
+
+It is important that the unmanaged calling convention declared in P/Invoke declaration matches the unmanaged calling convention
+used by the native implementation. Unmanaged calling convention mismatches lead to data corruptions and fatal crashes that require
+low-level debugging skills to diagnose.
+
+## Platform default calling convention
+
+Most platforms use one canonical calling conventions and explicitly specified calling convention is unnecessary in most cases.
+
+For x86 architecture, the default calling convention is platform-specific. StdCall is the default calling convention on Windows x86
```

Comment:
```suggestion
For the x86 architecture, the default calling convention is platform-specific. Stdcall ("standard call") is the default calling convention on Windows x86
```

Reply:
Do we want to mention the name mangling between stdcall and cdecl? For example `Foo@4` vs `_Foo`. That tends to be something people often get caught up on.

Reply:
I think it is enough to just give people a few hints that calling conventions are a problem on Windows x86. This is not meant to be Windows x86 calling convention troubleshooting doc. I would not mind reducing the Windows x86 specific coverage even more.

Both wikipedia and stackoverflow questions have coverage for the Windows x86 calling convention name mangling issues.